src/plugins/hiddenColumns/test/hiddenColumns.e2e.js

```diff
@@ -26,192 +33,2389 @@ describe('HiddenColumns', () => {
+          handsontable({
+            data: Handsontable.helper.createSpreadsheetData(2, 5),
+            rowHeaders: true,
+            colHeaders: true,
+            contextMenu: [CONTEXTMENU_ITEM_SHOW],
+            hiddenColumns: {
+              columns: [3, 4],
+            },
+          });
+
+          expect(spec().$container.find('tr:eq(0) th').length).toBe(4);
+          expect(spec().$container.find('tr:eq(1) td').length).toBe(3);
+          expect(getCell(0, 0).innerText).toBe('A1');
+          expect(getCell(0, 1).innerText).toBe('B1');
+          expect(getCell(0, 2).innerText).toBe('C1');
+          expect(getCell(0, 3)).toBe(null);
+          expect(getCell(0, 4)).toBe(null);
+
+          selectColumns(2);
+
+          contextMenu();
+          getPlugin('contextMenu').executeCommand(CONTEXTMENU_ITEM_SHOW);
+
+          expect(spec().$container.find('tr:eq(0) th').length).toBe(6);
+          expect(spec().$container.find('tr:eq(1) td').length).toBe(5);
+          expect(getCell(0, 0).innerText).toBe('A1');
+          expect(getCell(0, 1).innerText).toBe('B1');
+          expect(getCell(0, 2).innerText).toBe('C1');
+          expect(getCell(0, 3).innerText).toBe('D1');
+          expect(getCell(0, 4).innerText).toBe('E1');
+          expect(getSelected()).toEqual([[0, 2, 1, 4]]);
+          expect(`
+          |   ║   :   : * : * : * |
+          |===:===:===:===:===:===|
+          | - ║   :   : A : 0 : 0 |
+          | - ║   :   : 0 : 0 : 0 |
+          `).toBeMatchToSelectionPattern();
+        });
+      });
     });
 
-    expect(hot.getPlugin('hiddenColumns').enabled).toEqual(true);
-    expect($('.beforeHiddenColumn').size()).toBeGreaterThan(0);
+    describe('Changing selection after alter actions from context-menu', () => {
+      describe('should keep the row selection in the same position as before inserting the row', () => {
+        describe('above the selected row', () => {
+          it('when the first column is hidden', () => {
+            handsontable({
+              data: createSpreadsheetData(4, 4),
+              contextMenu: true,
+              rowHeaders: true,
+              colHeaders: true,
+              hiddenColumns: {
+                columns: [0]
+              },
+            });
+
+            const header = $('.ht_clone_left .htCore')
+              .find('tbody')
+              .find('th')
+              .eq(0);
+            simulateClick(header, 'RMB');
+            contextMenu(header);
+
+            $('.htContextMenu .ht_master .htCore')
+              .find('tbody td')
+              .not('.htSeparator')
+              .eq(0)
+              .simulate('mousedown')
+              .simulate('mouseup'); // Insert row above
+
+            expect(getSelected()).toEqual([[1, 0, 1, 3]]);
+            expect(getSelectedRangeLast()?.highlight?.row).toBe(1);
+            expect(getSelectedRangeLast()?.highlight?.col).toBe(1);
+            expect(getSelectedRangeLast()?.from?.row).toBe(1);
```

Comment:
Clarifying, I assume that there may be no selection, and the test won't pass. I thought that exception may break tests which is problematic. I have checked that on one test and thrown `TypeError` won't do that. The test will show `TypeError: Cannot read property 'highlight' of undefined` which is also good information for me. 

It's hard to understand your point of view as within `toBe` there is a concrete number. It would just say more precisely which test hasn't passed. However, I can see that @jansiegel also would like to get this change so I can do that.

Reply:
In that case for me, the optional chain is overused. IMO It's equlivement to something like
```
expect((function () {
  try {
    return getSelectedRangeLast().highlight.row
  } catch {}
}()))
```
When the `getSelectedRangeLast()` returns `undefined` that means that something really terrible was happened. The error shouldn't be silenced.

Reply:
Yeah, I agree with @budnix - with the chaining the error would tell us that there might be a problem with a wrong index being passed somewhere. Without the chaining we would have clear information, that there's a bigger problem there.

Reply:
> In that case for me, the optional chain is overused. IMO It's equlivement to something like
> 
> ```js
> expect((function () {
>   try {
>     return getSelectedRangeLast().highlight.row
>   } catch {}
> }())).toBe(1)
> ```
> 
> When the `getSelectedRangeLast()` returns `undefined` that means that something really terrible was happened in the selection module. The error shouldn't be silenced.

This error **won't be silenced**, because `toBe` will behave similarly to `catch` using this nomenclature. Anyway, I don't see the bigger benefits of my solution, so I will remove it 👍

Reply:
> This error won't be silenced

I think it would be _disguised_ as a different error, and it could potentially be problematic.

Reply:
Done within https://github.com/handsontable/handsontable/pull/6547/commits/eb95d43c792c773f9404c99b7dbed9de994fa6b9.